src/main/java/scmspain/karyon/restrouter/transport/http/RouteInterceptorSupport.java

```diff
@@ -0,0 +1,74 @@
+package scmspain.karyon.restrouter.transport.http;
+
+import io.netty.buffer.ByteBuf;
+import io.reactivex.netty.protocol.http.server.HttpServerRequest;
+import io.reactivex.netty.protocol.http.server.HttpServerResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Created by borja.vazquez on 25/9/15.
+ */
+public class RouteInterceptorSupport {
+  private static final Logger logger = LoggerFactory.getLogger(RouteInterceptorSupport.class);
+
+  private List<RouteOutInterceptor> routeOutInterceptorList = new ArrayList<>();
+
+  public RouteInterceptorSupport() {
+    routeOutInterceptorList.add((result, request, response) -> result.flatMap(value -> {
+      if (value == null) {
+        return Observable.empty();
+      } else {
+        String message = "Cannot handle value of type as Observable type in endpoint return: " + value.getClass();
+        logger.error(message);
+        return Observable.error(new RuntimeException(message));
+      }
+    }));
+  }
+
+  public void addOutInterceptor(RouteOutInterceptor routeOutInterceptor) {
+    routeOutInterceptorList.add(routeOutInterceptorList.size()-1, routeOutInterceptor);
+  }
+
+  /**
+   * It executes all interceptor in order to the response body.
+   * Return generic type will be Void for sure, because we add an interceptor at the end that will
+   * convert the value to a Void.
+   * @param responseBodyObs the responseBody Observable
+   * @param request the request
+   * @param response the response
+   * @return the Void observable that karyon expects
+   */
+  @SuppressWarnings("unchecked")
+  public Observable<Void> execute(Observable<?> responseBodyObs, HttpServerRequest<ByteBuf> request, HttpServerResponse<ByteBuf> response) {
+    return (Observable<Void>)applyInterceptors(responseBodyObs, request, response, 0);
```

Comment:
I think we are reinventing the wheel, this class is doing the same thing that a EventDispatcher or if you think in a reactive way a Subject.
Personally, my idea of doing this is launching events when it will be useful and the subject will do their job sending to subscribers.
There is no need to return "Observable\<Object\>" in the endpoint because the endpoint only need to notify that already finished their job with an event.
With my approach will could have "interceptor asyncronous and syncronous" depending of what you want to do in the event and it would be easy to implement only changing the schedulers strategy :)